Sources/Sentry/SentrySpan.m

```diff
@@ -15,6 +15,7 @@
 
 @implementation SentrySpan {
     NSMutableDictionary<NSString *, id> *_extras;
+    NSMutableDictionary<NSString *, id> *_tags;
```

Comment:
`h`: I just realized, that we already have `tags` in the `SpanContext`. I think all we need is to forward the method calls to the `context`.

Reply:
`SpanContext` is supposed to be immutable though:

i.e: [C#](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/SpanContext.cs)

To my surprise in Java it's mutable, which means you could give a span context to a method to start a span, that method would rely on it to decide on sampling and data magically changes.

Unless I'm missing something, that's the goal of splitting Span and SpanContext (even though there seems to be no documentation about this on develop.sentry.dev). Otherwise what's the point of having both Span and SpanContext?

Reply:
@bruno-garcia point it out that SpanContext should be immutable, which it is not, the `sample` property and `tags` property can be mutate. 

The question now is, should we fix the project to make SpanContext immutable, some work required here, or we leave it as it is.

I like .NET approach. Span does not have a context, it implements SpanContext.

Reply:
We can't change that right now, because it would be a breaking change. I created an issue for that: https://github.com/getsentry/sentry-cocoa/issues/1254. For now, I propose doing the same as in Java. We forward calls from `Span` to `SpanContext` to have a single source of truth.

Reply:
As discussed with @bruno-garcia, the `Span` shouldn't modify the `SpanContext`. Let's keep two dictionaries and merge them in serialize.